propagator/xray/lib/opentelemetry/propagator/id_generator.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Propagator
+    module XRay
+      # This module is intended to only be used as an override for how to generate IDs to be compliant with XRay
+      module IDGenerator
+        extend self
+
+        # Generates a valid trace identifier, a 16-byte string with at least one
+        # non-zero byte.
+        # AWS Docs: https://docs.aws.amazon.com/xray/latest/api/API_PutTraceSegments.html
+        # hi - 4 bytes timestamp, 4 bytes random (Mid)
+        # low - 8 bytes random.
+        # Since we include timestamp, impossible to be invalid.
+        # @return [bytes] a valid trace ID that is compliant with AWS XRay.
+        def generate_trace_id
+          time_hi = generate_time_bytes
+          low_and_low = RANDOM.bytes(12)
+          time_hi << low_and_low
```

Comment:
For consistency with the method doc:
```suggestion
          mid_and_low = RANDOM.bytes(12)
          time_hi << mid_and_low
```